FluxNotes/flux
# Stop medication structured phrase

This PR address jira issue 1139 to create a new shortcut for #stop medication. Worked with @gregquinn2001 and @daniellee1025 on note parser updates.

**To test this PR** 
The web app:

Open a new note and type #stop medication. Then enter @active medication and choose one of the medications from the list. View the medication charts in the targeted data panel. A medication that has been stopped will not display the chart and the text updates accordingly. See below where the medication chosen was Lisinopril:
![image](https://user-images.githubusercontent.com/26877672/42048377-16bb4f3a-7ad1-11e8-9644-70f3d442ccd8.png)

This also works when you change the medication type on a medication that already has a medication change. For example updating PACLitaxel from reduced to stopped will update the patient and the visualization.

Note parser:

Note 14 was added for the #stop medication shortcut.

Run the following command:
'yarn parse src/noteparser/samples/note14.txt'
The results of parsing the note will be displayed. You will see a medication change object with the type set to "stop" and a flux medication requested of "ibuprofen 600mg tablet"


**Changes made**
- Added #stop medication shortcut and @active medication shortcut
- Updated shortcuts.json (added new attributes) and note parser to support these new shortcuts (*Note: Added infrastructure to support structured phrases using # followed by @ and support parsing of inserters)
- Updated the note parser to handle inserters and added a patient to be able to look up active medications on the patient when parsing @active medication
- Added a new sample note with #stop medication and @active medication 
- Fixed broken backend note parser tests (updated expected jsons to include missing attributes and removed entryid and shrid prior to making comparison)
- Updated Flux objects to set entry information when none exists and added setters 
- Updated MedicationRangeChartVisualizer to change display when medication change is type "stop"
- Refactored some code in SummaryMetadata to more generically handle medication changes



_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._




## Submitter:

**Running the Application**

- [x] Manually tested in Chrome
- [x] Manually tested in IE

**Documentation**

- N/A Cheat sheet is updated
- N/A Demo script is updated 
- [x] Documentation on Wiki has been updated 
- N/A Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- [x] Note parser has been updated if structured phrases change

**Code Quality**

- [x] 4-space indents - convert any tabs to spaces
- [x] Use public class field syntax for binding functions - ex. handleChange = () => { ... };

**Tests**

- N/A Added Enzyme-UI tests for slim mode 
- N/A Added Enzyme-UI tests for full mode
- N/A Added backend tests for new functionality added
- [x] Added note parser examples to test new functionality


## Reviewer 1: Name



## Reviewer 2: Name
